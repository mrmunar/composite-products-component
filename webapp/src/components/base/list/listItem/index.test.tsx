import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import ListItem from './';

afterEach(cleanup);

const testData = { id: 1, name: 'item1' };

test('should take a snapshot', () => {
    const { baseElement } = render(<ListItem item={testData} />);
    expect(baseElement).toMatchSnapshot();
});

test('should render an item', () => {
    const { getByTestId } = render(<ListItem item={testData} />);
    expect(getByTestId('list-item')).toHaveTextContent('item1');
});

test('should show item as link', () => {
    const { getByTestId } = render(<Router><ListItem item={testData} withLinks /></Router>);
    expect(getByTestId('list-item-link')).toHaveTextContent('item1');
});


