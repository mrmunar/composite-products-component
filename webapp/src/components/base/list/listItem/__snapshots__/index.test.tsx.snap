// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <li
      class="list-group-item"
      data-testid="list-item"
    >
      item1
    </li>
  </div>
</body>
`;
